### ISSUE TITLE
Converting standalone Graylog server to cluster setup causes global inputs to fail

### ISSUE THREAD

**Issue by emanzaman** [Graylog2/graylog2-server#5036]
I have total three nodes A, B and C.
Initially I was running graylog only on node A. Now, I am trying to have a graylog cluster with all three nodes A, B and C. 
This also works fine I can see the nodes belognign to graylog cluster as shown below:
![image](https://user-images.githubusercontent.com/38486100/44841562-3ec98700-ac61-11e8-9e33-80edfb3512d0.png)

## Expected Behavior
If global inputs were working in standalone mode it should also work in cluster setup.

## Current Behavior
But the global inputs fails to start on the new nodes B and C and complains that the address is already in use as shown below:
![image](https://user-images.githubusercontent.com/38486100/44841620-6d476200-ac61-11e8-849d-dc4ec1c85c59.png)

## Your Environment

* Graylog Version: 2.4.3
* Elasticsearch Version: 5.6.2
* MongoDB Version: 3.6.2
* Operating System: Red Hat
* Browser version: Chrome

**Comment by emanzaman** [comment]
Yes I confirmed that no process on nodes B and C was using 12205 port. Even after graylog has been started on nodes B and C, the port is shown to be used by graylog process itself. So wanted to ask if I was missing any configuration?

`$> netstat -nlp | grep 12205
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
tcp        0      0 0.0.0.0:12205               0.0.0.0:*                   LISTEN      7586/java
$> ps -ef|grep graylog
rsuurun   7586     1 34 11:30 ?        00:01:10 /opt/app/ignite/pkgs/pd/jdk/1.8.0_66-64/bin/java -Xloggc:/apps/IRDuatrun/rsu/ldn/dev/graylog-server/logs/lonrs10753/graylog-server-gc.log -Dlog4j.configurationFile=/apps/IRDuatrun/rsu/ldn/dev/graylog-server/etc/config/log4j2.xml -Xms2g -Xmx2g -XX:NewRatio=1 -server -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintTenuringDistribution -XX:+PrintGCApplicationStoppedTime -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=128M -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/apps/IRDuatrun/rsu/ldn/dev/graylog-server/logs/lonrs10753 -DlogsDir=/apps/IRDuatrun/rsu/ldn/dev/graylog-server/logs/lonrs10753 -XX:OnOutOfMemoryError='/apps/IRDuatrun/rsu/ldn/dev/graylog-server/bin/killOnOOM.sh' -jar graylog.jar server -f /apps/IRDuatrun/rsu/ldn/dev/graylog-server/etc/config/graylog-server.ldn.dev.conf -p /apps/IRDuatrun/rsu/ldn/dev/graylog-server/logs/lonrs10753/graylog-server-service.pid`

**Comment by dennisoelkers** [comment]
Do you have any other inputs configured? What happens when you stop and restart the global input?

**Comment by emanzaman** [comment]
I just have two global inputs one on port 12205 and the other on port 5556. Both of these inputs fail on cluster setup. 
On restarting global inputs, the two inputs on nodes B and C fail

**Comment by emanzaman** [comment]
I just restarted all my graylog servers and now the global inputs are running fine on the same ports.
![image](https://user-images.githubusercontent.com/38486100/44848448-66761a80-ac74-11e8-8b43-02f16811ce06.png)

Could you please direct me to any possible wrong configuration that I have done? Because this is the test environment and we dont want any such issues on Production.
Thanks a lot

**Comment by dennisoelkers** [comment]
I cannot see any wrong configuration that you did. It seems to be a bug triggered by an "edge case" (I am a bit apologetic here) of global inputs being launched because of nodes joining the cluster. It would help us a lot if you would test if that behavior is reproducible. That means: you could delete the input, shut down nodes B & C, configure the input again (maybe with a different port just to be sure) and start nodes B & C again.

**Comment by emanzaman** [comment]
Yes I am able to reproduce it by following these steps:

1. Deleted all the data (mongodb, elasticsearch and graylog) on nodes A, B and C.
2. Started only one graylog instance on node A (there is no cluster even for elasticsearch and mongodb)
3. Then created a cluster for mongo & elasticsearch and started the graylog nodes.
4. The global inputs doesn't start as shown below:
![image](https://user-images.githubusercontent.com/38486100/44853669-2cad1000-ac84-11e8-9824-d01afad6def0.png)

Please let me know if there is any workaround we can do to avoid this because our release will not be accepted if anything is failing on startup.

Thanks a lot.

**Comment by jalogisch** [comment]
For me this is not reproducable. It looks like something that happens in your local environment. But is no error within Graylog.

I had setup one Graylog, based on CentOS7 following our Step-by-step Guide. Created one global input in that environment. After everything was running successfull I started a new CentOS7 VM and installed Graylog again following the same steps. But used the same MongoDB and Elasticsearch as with the first box - having the same `password_secret` and `root_password_sha2` on both nodes. 

After the startup the second box joinded the cluster and started the previous configured global inputs.

Please describe *exactly* how we can reproduce this with all needed information, include as much information as possible. That would include system automation recipies, scripts and everything - because following standard methods it is not possible to reproduce.

**Comment by emanzaman** [comment]
@jalogisch ,
I am not sure if I can share the scripts but we are running Graylog using manual setup and we invoke Graylog using graylog/2.4.3/bin/graylogctl and following 
 the steps mentioned in http://docs.graylog.org/en/2.4/pages/installation/manual_setup.html

Noticed that you have just started a second instance to form a cluster but my requirement is to restart all the servers in order to form a cluster.
